### PR TITLE
Injection of Permission checks into OOB ISI.

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: iroha/src/asset.rs [hunk 1]
```diff
@@ -207,7 +222,17 @@ pub mod isi {
     }
 
     impl Mint<Asset, u128> {
-        fn execute(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
+        fn execute(
+            &self,
+            authority: <Account as Identifiable>::Id,
+            world_state_view: &mut WorldStateView,
+        ) -> Result<(), String> {
+            PermissionInstruction::CanMintAsset(
```
Comment (takemiyamakoto) [inline iroha/src/asset.rs:230]: Does this consider the ability to mint a specific asset, rather than all assets?

Comment (humb1t) [reply iroha/src/asset.rs:230]: It consider the ability to Mint `Asset` of the specific `AssetDefinition`:
```rust
230             PermissionInstruction::CanMintAsset(
231                 authority,
232                 self.destination_id.definition_id.clone(),
233                 None,
234             )
```

File: iroha/src/lib.rs [hunk 2]
```diff
@@ -83,9 +83,36 @@ impl Iroha {
...
+        let mut accounts = HashMap::new();
+        accounts.insert(account_id, account);
+        let domain = Domain {
+            name: domain_name.clone(),
+            accounts,
+            asset_definitions,
+        };
+        let mut domains = HashMap::new();
+        domains.insert(domain_name, domain);
+        let world_state_view = Arc::new(RwLock::new(WorldStateView::new(Peer::with_domains(
```
Comment (eadventurous) [inline iroha/src/lib.rs:112]: I think this should be in genesis block. Right now it seems to be not in blockchain at all.

If not in genesis block, then maybe at least take these values from configuration?

Comment (humb1t) [reply iroha/src/lib.rs:112]: @eadventurous totally agree with you, let's put in on an additional issue for a genesis block functionality (maybe some things we will move to another config, but we definitely need genesis)
image
Jira is not working right now, I will hold PR till issue creation

Comment (humb1t) [reply iroha/src/lib.rs:112]: https://jira.hyperledger.org/secure/RapidBoard.jspa?rapidView=264&view=detail&selectedIssue=IR-756

**Pull request merged**